src/Controllers/AddProductController.php

```diff
@@ -23,6 +24,82 @@ public function __construct($productModel)
 
     public function __invoke(Request $request, Response $response, array $args)
     {
-        return $response;
+        $newProductData = $request->getParsedBody();
+
+        try {
+            $newProduct = new ProductEntity(
+                $newProductData['sku'],
+                $newProductData['name'],
+                $newProductData['price'],
+                $newProductData['stock']);
+
+        } catch (\Throwable $e) {
+            $responseData = [
+                'success' => false,
+                'message' => $e,
+                'data' => []
+            ];
+
+            $response->getBody()->write(json_encode($responseData));
+            $response->withStatus(400);
+
+            return $response->withHeader('Content-Type', 'application/json');
+        }
+
+        try {
+            $productExists = $this->productModel->checkProductExists($newProduct->getSku());
+
+            if ($productExists) {
+                $responseData = [
+                    'success' => false,
+                    'message' =>
+                        'This product already exists in the database. Either update the old product or use a new SKU.',
+                    'data' => []
+                ];
+                $response->getBody()->write(json_encode($responseData));
+                $response->withStatus(400);
+
+                return $response->withHeader('Content-Type', 'application/json');
+
+            } else {
+                $query_success = $this->productModel->addProduct($newProduct);
+
+                if ($query_success) {
+                    $responseData = [
+                        'success' => true,
+                        'message' =>
+                            'Product successfully added.',
+                        'data' => []
+                    ];
+                    $response->getBody()->write(json_encode($responseData));
+                    $response->withStatus(200);
+
+                    return $response->withHeader('Content-Type', 'application/json');
+
+                } else {
```

Comment:
you can get rid of the else bit and just have this run after the if ($query_success) bit, since it now returns if it $query_success = true; and won't reach this code